### DISCUSSION TITLE
Problems with GifDisposalMethod in converting individual GIF frames to a list of PNGs

### DISCUSSION THREAD

**Discussion by bigfella2** [SixLabors/ImageSharp#2584] [Q&A] [answered]
Hello,

I have a GIF that uses the NotDispose GifDisposalMethod, and whenever I attempt to write all of the frames' pixel data to new images and save them as PNGs, all of the frames after the first frame lose the previous frame's data wherever there are transparent pixels. I know that this is an issue involving the GifDisposalMethod, but is there a way to "coalesce" the GIF before attempting to save the individual frames to PNG so that each frame contains all of the information?

Thank you.

**Comment by JimBobSquarePants** [comment, marked as answer]
Sorry yeah. I broke the gif decoder accidentally with v3 by removing coalescing. 

I’m currently working on a rewrite of the gif decoder and encoder which I will have merged in a few days. This fixes all known issues. Once merged you can grab it from the nightly feed.

**Reply by JimBobSquarePants** [reply]
Merged now
